tests/test_performance.py

```
@@ -0,0 +1,569 @@
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from unittest.mock import Mock, patch, AsyncMock
+from typing import List, Dict, Any
+import gc
+import sys
+
+from tidal_mcp.service import TidalService
+from tidal_mcp.auth import TidalAuth
+from tidal_mcp.models import Track, Album, Artist, Playlist, SearchResults
+from tidal_mcp.utils import (
+    sanitize_query, format_duration, validate_tidal_id,
+    calculate_playlist_stats
```

Ruff (0.12.2)

`tidal_mcp.utils.calculate_playlist_stats` imported but unused

Remove unused import: `tidal_mcp.utils.calculate_playlist_stats`

(F401)